### DISCUSSION TITLE
Gaussian pulse excitation fine control

### DISCUSSION THREAD

**Discussion by oberstet** [thliebig/openEMS-Project#148] [Q&A Usage]
I've done some experiments rendering (in Paraview) the current distribution in a helical antenna (like https://docs.openems.de/python/openEMS/Tutorials/Helical_Antenna.html) excited with a Gaussian pulse (in openEMS/Python).

Here is a picture showing current in helix for a number of time steps during the excitation (time from top to bottom):

![test4](https://github.com/thliebig/openEMS-Project/assets/233340/b0912be5-076b-4611-8937-44296c436499)

movie: https://www.youtube.com/watch?v=pRFfFKI08O0

----

I am using openEMS **SetGaussExcite**, and I am wondering if I can control the detailed pulse shape in terms of *excitation time* or *number of center frequency cycles*.

What can be seen in the rendered current distribution is that the pulse envelope, as it travels along the helix, almost or actually takes longer than the helix extends, and I am worried about the interaction with the reflection from the helix top end. I've analyzed this using E-field which shows the reflections and change of orientation for the downward traveling reflection (https://www.youtube.com/shorts/2anvkk9EtlE)

Can I e.g. generate a pulse consisting of exactly 3 or 5 or 7 center frequency cycles, while manually providing the peak amplitude of each cycle .. I can manually decay those from the mid cycle to the others.

Is that possible? I can only see 2 parameters in `SetGaussExcite`

```
Set a Gaussian pulse as excitation signal.
        f0 – float – Center frequency in Hz.
        fc – float – -20dB bandwidth in Hz.
```

obviously I know the center frequency and the helix length, so I could control timing (when the envelope peak hits the helix top) and such.

also, I've had a quick look at matlab and what is provides

```
tc = gauspuls('cutoff',fc,bw,bwr,tpe) returns the cutoff time tc at which the trailing pulse envelope falls below tpe dB with respect to the peak envelope amplitude.
```
source: https://de.mathworks.com/help/signal/ref/gauspuls.html

Is something similar available in openEMS, how are people analyzing/tuning this in antennas?

**Comment by KJ7LNW** [comment]
Have you looked at [`SetCustomExcite`](https://github.com/thliebig/openEMS/blob/master/matlab/SetCustomExcite.m)?

```octave
% function FDTD = SetCustomExcite(FDTD,f0,funcStr)
%
% f0 : nyquist rate
% funcStr : string describing the excitation function e(t)
%
% see also SetSinusExcite SetGaussExcite
%
% e.g for a ramped sinus excite...
% T = 1/f0;
% FDTD = SetCustomExcite(FDTD,1e9,..
% [ '(1-exp(-1*(t/' num2str(T) ')^2) ) * sin(2*pi*' num2str(f0) '*t)' ]);
%
% openEMS matlab interface
% -----------------------
% author: Thorsten Liebig
```

**Comment by oberstet** [comment]
thanks for responding!

yeah, I've looked at it, it is currently missing exposing a Python entry point, which I've added here

https://github.com/thliebig/openEMS/pull/129/files

side note: I'm also not convinced (from a SW eng PoV) that I want to learn/dig a new syntax of some ad-hoc flavor just to add a function;) I do have Python and C++ around, so why do I need another thing? anyways, that's SW eng related, not RF

also, above PR has more things, including a skeleton for a new `SetDumpedExcite` as well as a new `SetupNoiseExcite`.

actually at his stage the PR was only intended to get some first feedback and measure interest.

now, the feedback on the PR hasn't been exactly positive, or I should say: basically multiple RF experts have told me that such a continuous white noise excite isn't what is commonly done in RF, and that it wouldn't make much sense.

the PR has these other 2 bits, the Python exposing, and also the load previously dumped excite thing, but there doesn't seem to be lots of interest.

my own interest is also kinda postponed on this issue, because last weeks I learned a lot, including what will be my concrete RF design problem:

1. decreasing size 2x, increasing bandwidth 6x vs the current antenna model I have
2. actually measuring this in real world, because I now have [multiple models of a single antenna which are all "more or less good"](https://github.com/thliebig/openEMS-Project/discussions/160#discussioncomment-7856959), and which I could each optimize fully on their own, but it's pointless without knowing which one is actually closest to reality.

the antenna bandwidth problem is the real problem. I want a sub-ghz antenna for a handheld device with 600mhz bandwidth;)

**Reply by KJ7LNW** [reply]
> yeah, I've looked at it, it is currently missing exposing a Python entry point, which I've added here

neat!

> actually at his stage the PR was only intended to get some first feedback and measure interest.

even if white noise is not the solution, I hope the pull request gets pulled in if it adds features that were previously lacking.

> decreasing size 2x, increasing bandwidth 6x vs the current antenna model I have

Assuming you mean on a helical antenna:

My son and I read a few papers on how to scale bandwidth on helical antennas based on the spacing between turns any papers give overview. Of course, you have to simulate for your exact model, but it gives an idea on what metrics affect what behavior for bandwidth and gain.

I recommend that you read this currently papers linked part-way-down this page:

 - https://www.qrz.com/db/KJ7NLL

**Comment by oberstet** [comment]
> My son and ..

he's pretty cool for his age;) tbh, from your very 1st post, I immediately visited YT, and saw him talking, and I (wrongly) thought your reply _here_ would have been him, and I thought: alright! then I figured, this really can't be, and I found your blog/webpage which revealed the detail of father+son;) anyways, cool.

back to the topic: thanks for your links rgd helical antennas! will read. tbh, I don't have big expectations at this point that helical alone would solve my problem. but of course I need to simulate.

I am currently cleaning up my code, prepared batch simulation, need to add a database, etc .. but "planar helical" is next, I mean I have that, but not yet with correct meshing. I do have correct meshing for dipole, monopole and planar monopole, and the latter is what my numbers rgd 2x size and 6x bandwidth stem from.
